backend/schedulerd/executor.go

```diff
@@ -355,6 +354,13 @@ func buildRequest(check *types.CheckConfig, s store.Store) (*types.CheckRequest,
 		for _, hook := range hooks {
 			if hookIsRelevant(hook, check) {
 				request.Hooks = append(request.Hooks, *hook)
+				if len(hook.RuntimeAssets) != 0 {
+					for _, asset := range assets {
+						if assetIsRelevant(asset, hook.RuntimeAssets) {
+							request.Assets = append(request.Assets, *asset)
+						}
+					}
+				}
```

Comment:
Could flattening all of our assets lead to some unexpected behaviour? 

as an example my check defines `RuntimeAssets = ["python2"]` and one of my hooks defines `RuntimeAssets = ["python3"]`

Reply:
Both of those assets would be loaded into the check request, however the execution environment for the check and hook should be different.

Reply:
Actually, there is some logic I can add to ensure this!

Reply:
@jamesdphillips I added a map `HookAssets` to the `CheckRequest` to ensure only the assets required for each check and hook are loaded into the environment. I'd love to get a gut check on that change!